subsys/bluetooth/mesh/net.c

```diff
@@ -88,7 +88,7 @@ struct bt_mesh_net bt_mesh = {
 	},
 };
 
-static uint32_t dup_cache[4];
+static uint32_t dup_cache[CONFIG_BT_MESH_MSG_CACHE_SIZE];
```

Comment:
The dup cache is not the same as the message cache, but it's an interesting change. 

In theory, we can replace the entire message cache with this, and save a lot of deobfuscation, although it'll emit false positives for one out of every `4.29 billion / cache_size` packets. I believe this was the original motivation behind having a "proper" message cache, but it seems like an acceptable downside to me. I can make a separate PR after this one if @jhedberg agrees

Reply:
Yes, I understand that it's not the same. I was hoping for a discussion about this change. Initially I though this was left out by accident and I wanted to remove it since we have proper message cache. Then I ran the test with this small change and it fixed the test so I figured we will talk about it in the comments.

Reply:
Interesting. The intent is that this would filter out most duplicate messages, so we don't have to run deobfuscation over and over again, only to discover that the message is a duplicate. Sounds like there's a hole in the message cache implementation then, and this change hides it. The way I see it, we should either replace the message cache completely or fix it. I'm leaning towards the former 😄

Reply:
It seems like this discussion wasn't yet resolved? What's the value for `CONFIG_BT_MESH_MSG_CACHE_SIZE` that you're using? I'd like to understand better why any of this is causing problems, since the purpose of `dup_cache` is just to eliminate unnecessary processing, i.e. the stack should work whether it exists or not, and regardless of what size it is.

Reply:
That said, obviously there is a risk for collisions since we're hashing a message into just 32 bits, however I'd like to understand better why the test case was failing.

Reply:
I looked at it once more and I think I messed up a bit. At first I was testing with CONFIG_BT_MESH_MSG_CACHE_SIZE=2 because that was the default value in PTS IXIT. It failed (more messages were ignored than PTS expected) so I made this change and it worked. Then I tested with size=10 and it worked again. Now I tried with size=10 but without the fix and it works. So it works when size >= dup_cache size which is understandable. Still I think that we should have a define or a config value for dup_cache size as this magic value (4) can be confusing.

Reply:
I see. I think the collision risk with the 32bit hash is acceptable, given that we can completely avoid doing deobfuscation on duplicate packets. Even at an rx rate of 100 new packets per second and a cache size of 10, we'll be dropping a valid packet only once every 50 days. The average packet drop rate from network interference is orders of magnitude higher, so it wouldn't be noticeable in the application layer. 

I think the current change is the right one, but as it makes the message cache redundant, we should additionally remove that completely.

Reply:
Discussed with @jhedberg offline:
Okay to remove the existing message cache functionality, and rely entirely on the dup_cache. 
@michal-narajowski I think this should be done as part of this PR, as it already makes the existing message cache functionality redundant.